# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.h
# Commit: b2727a598452
# Full Hash: b2727a598452fb942e0611df5f931d4df0efaceb
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2024-01-18 04:47:51
# Regressor Bug: 1765615
# File Overlap Count: 1
# Description:
#   Bug 1765615 - Handle most changes to CSS `contain` and `content-visibility` without needing to reconstruct frames. r=layout-reviewers,emilio
#   
#   Right now, we reconstruct frames in response to a change in the CSS
#   `contain` property or `content-visibility`. This patch tries to optimize
#   this a bit:
# ==============================================================================

diff -r 5608664350d8 -r b2727a598452 layout/base/PresShell.h
--- a/layout/base/PresShell.h	Wed Jan 17 07:22:29 2024 +0000
+++ b/layout/base/PresShell.h	Wed Jan 17 08:22:48 2024 +0000
@@ -1748,6 +1748,9 @@
   void RegisterContentVisibilityAutoFrame(nsIFrame* aFrame) {
     mContentVisibilityAutoFrames.Insert(aFrame);
   }
+  void UnregisterContentVisibilityAutoFrame(nsIFrame* aFrame) {
+    mContentVisibilityAutoFrames.Remove(aFrame);
+  }
   bool HasContentVisibilityAutoFrames() const {
     return !mContentVisibilityAutoFrames.IsEmpty();
   }